NPE in KotlinWhenStringFilter

### Steps to reproduce

JaCoCo version: master (commit 8b586a98803923f11a9b05fe214fa25c2c23207f) 
Operating system: OSX
Tool integration: CLI

Run

    java -jar jacococli.jar report --classfiles NPE.zip

on the attached class file.

### Expected behaviour

Command terminates without exception.

### Actual behaviour

```
Exception in thread "main" java.io.IOException: Error while analyzing NPE.zip@$value$ReadableByteChannel.class.
	at org.jacoco.core.analysis.Analyzer.analyzerError(Analyzer.java:161)
	at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:133)
	at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:156)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:192)
	at org.jacoco.core.analysis.Analyzer.analyzeZip(Analyzer.java:264)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:195)
	at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:225)
	at org.jacoco.cli.internal.commands.Report.analyze(Report.java:109)
	at org.jacoco.cli.internal.commands.Report.execute(Report.java:83)
	at org.jacoco.cli.internal.Main.execute(Main.java:89)
	at org.jacoco.cli.internal.Main.main(Main.java:104)
Caused by: java.lang.NullPointerException
	at org.jacoco.core.internal.analysis.MethodCoverageCalculator.ignore(MethodCoverageCalculator.java:159)
	at org.jacoco.core.internal.analysis.filter.KotlinWhenStringFilter$Matcher.match(KotlinWhenStringFilter.java:98)
	at org.jacoco.core.internal.analysis.filter.KotlinWhenStringFilter.filter(KotlinWhenStringFilter.java:37)
	at org.jacoco.core.internal.analysis.filter.Filters.filter(Filters.java:57)
	at org.jacoco.core.internal.analysis.ClassAnalyzer.addMethodCoverage(ClassAnalyzer.java:118)
	at org.jacoco.core.internal.analysis.ClassAnalyzer.access$2(ClassAnalyzer.java:113)
	at org.jacoco.core.internal.analysis.ClassAnalyzer$1.accept(ClassAnalyzer.java:107)
	at org.jacoco.core.internal.flow.ClassProbesAdapter$2.visitEnd(ClassProbesAdapter.java:89)
	at org.objectweb.asm.ClassReader.readMethod(ClassReader.java:1287)
	at org.objectweb.asm.ClassReader.accept(ClassReader.java:688)
	at org.objectweb.asm.ClassReader.accept(ClassReader.java:400)
	at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:115)
	at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:131)
	... 9 more
```
[NPE.zip](https://github.com/jacoco/jacoco/files/3636164/NPE.zip)

--- a/org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinWhenStringFilterTest.java
+++ b/org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinWhenStringFilterTest.java
@@ -105,4 +105,21 @@ public class KotlinWhenStringFilterTest extends FilterTestBase {
 		assertIgnored(new Range(expectedFromInclusive, expectedToInclusive));
 	}
 
+	@Test
+	public void should_not_filter_empty_lookup_switch() {
+		final MethodNode m = new MethodNode(InstrSupport.ASM_API_VERSION, 0,
+				"name", "(Ljava/lang/String;)V", null, null);
+		m.visitVarInsn(Opcodes.ALOAD, 1);
+		m.visitMethodInsn(Opcodes.INVOKEVIRTUAL, "java/lang/String", "hashCode",
+				"()I", false);
+		final Label defaultCase = new Label();
+		m.visitLookupSwitchInsn(defaultCase, null, new Label[] {});
+		m.visitLabel(defaultCase);
+		m.visitInsn(Opcodes.RETURN);
+
+		filter.filter(m, context, output);
+
+		assertIgnored();
+	}
+
 }

--- a/org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinWhenStringFilter.java
+++ b/org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinWhenStringFilter.java
@@ -67,6 +67,10 @@ public final class KotlinWhenStringFilter implements IFilter {
 				hashCodes = tableSwitch.labels.size();
 			}
 
+			if (hashCodes == 0) {
+				return;
+			}
+
 			final Set<AbstractInsnNode> replacements = new HashSet<AbstractInsnNode>();
 			replacements.add(skipNonOpcodes(defaultLabel));
 

--- a/org.jacoco.doc/docroot/doc/changes.html
+++ b/org.jacoco.doc/docroot/doc/changes.html
@@ -47,6 +47,8 @@
   <li>To avoid failures with invalid class files report generation now checks
       that source references are actually files
       (GitHub <a href="https://github.com/jacoco/jacoco/issues/941">#941</a>).</li>
+  <li><code>NullPointerException</code> during filtering
+      (GitHub <a href="https://github.com/jacoco/jacoco/issues/942">#942</a>).</li>
 </ul>
 
 <h3>Non-functional Changes</h3>

